TrafficSimulator/Assets/Utilities/CarSpawner.cs

```diff
@@ -142,32 +148,72 @@ private void SpawnCars()
-
                 // Return if there are no cars to spawn
                 if (carsToSpawn == 0)
                     return;
 
                 // Calculate the offset
                 _laneNodeCurrent = _lanes[i].StartNode;
 
-                // Spawn cars
-                for (int j = 0; j < carsToSpawn; j++)
+                List<float> sections = DivideLaneToSections(_lanes[i]);
+                List<float> sectionRatios = CalculateSectionRatios(sections);
+                for (int j = 0; j < sections.Count; j++)
                 {
-                    // Check if max cars have been spawned
-                    if (_mode == SpawnMode.Total && _carCounter >= TotalCars)
-                        return;
+                    // Calculate the number of cars to spawn
+                    int carsToSpawnSection = Mathf.CeilToInt(carsToSpawn * sectionRatios[j]);
+                    for (int k = 0; k < carsToSpawnSection; k++)
+                    {
+                        // Check if max cars have been spawned
+                        if (_mode == SpawnMode.Total && _carCounter >= TotalCars)
+                            return;
+
+                        // Spawn car
+                        if(!_laneNodeCurrent.RoadNode.IsIntersection() && _laneNodeCurrent.RoadNode.Type != RoadNodeType.JunctionEdge && _laneNodeCurrent != null && !_laneNodeCurrent.HasVehicle() && _laneNodeCurrent.Next != null && _laneNodeCurrent.Next.Position != _laneNodeCurrent.Position)
+                        {
+                            SpawnCar(i);
+                            _carCounter++;
+                        }
+                        // Calculate the next spawn node in the section based on distance
+                        _offset += sections[j] / (carsToSpawnSection);
+                        _laneNodeCurrent = CalculateSpawnNode(_offset, _lanes[i]);
+                    }
+                }
+                _offset = 0;
+            }
+        }
 
-                    // Spawn individual car at current node
-                    SpawnCar(i);
+        /// <summary>Divides a lane into sections based on intersections</summary>
+        private List<float> DivideLaneToSections(Lane lane)
+        {
+            LaneNode curr = lane.StartNode;
+            LaneNode prev;
+            List<float> sections = new List<float>();
+            float sectionLength = 0;
 
-                    _carCounter++;
-                    _offset += _lanes[i].Length / carsToSpawn;
-                    _laneNodeCurrent = CalculateSpawnNode(_offset, _lanes[i]);
+            while (curr != null)
+            {
+                // Check if the current node is an intersection or if the next node is null to determine the end of a section
+                if(curr.RoadNode.IsIntersection() || (curr.RoadNode.Type == RoadNodeType.JunctionEdge) || curr.Next == null)
+                {
+                    sections.Add(sectionLength);
```

Comment:
An section between the intersection node and an junction node will be added currently. Since the code you wrote in Lane and this does pretty much the same thing, you could use a similar logic to the lane code.